src/main/java/janggi/piece/Piece.java

```diff
@@ -0,0 +1,70 @@
+package janggi.piece;
+
+import janggi.Movements;
+import janggi.Path;
+import janggi.Team;
+import janggi.board.Board;
+import janggi.board.Position;
+
+import janggi.board.PositionOutOfBoardBoundsException;
+import java.util.List;
+
+public abstract class Piece {
+    protected final Team team;
+
+    public Piece(Team team) {
+        this.team = team;
+    }
+
+    public boolean isSameTeam(Team team) {
+        return this.team == team;
+    }
+
+    public boolean isDifferentTeam(Team team) {
+        return this.team != team;
+    }
```

Comment:
이런 경우 아래와 같이 메서드를 엮어주는 것이 좋아요. 로직에 수정이 필요할 때 한 곳을 누락해서 버그가 발생하는 경우가 종종 발생하기도 하구요. 메서드 체이닝을 통해 두 메서드가 서로 관련있다고 의미를 부여할 수도 있어요.
```suggestion
    public boolean isSameTeam(Team team) {
        return this.team == team;
    }

    public boolean isDifferentTeam(Team team) {
        return !isSameTeam(team);
    }
```

Reply:
말씀해주신 부분과 ```Board#isPieceNotExsits```도 함께 수정했습니다.😊
730696b64398edee01b1905686e9fea3e499b683